File: backend/node-express/src/repositories/comment.repository.js
const { prisma } = require('../config/database');

class CommentRepository {
  async create(data) {
    return prisma.comment.create({ data });
  }

  async findByQuestionId(questionId) {
    return prisma.comment.findMany({
      where: { questionId },
      orderBy: { createdAt: 'asc' }
    });
  }

  async delete(id) {
    return prisma.comment.delete({ where: { id } });
  }
}

module.exports = new CommentRepository();

File: backend/dot-net/Repositories/CommentRepository.cs
using Microsoft.EntityFrameworkCore;
using backend.Data;
using backend.Models;

namespace backend.Repositories;

public class CommentRepository
{
    private readonly AppDbContext _context;

    public CommentRepository(AppDbContext context)
    {
        _context = context;
    }

    public Comment Create(Comment comment)
    {
        _context.Comments.Add(comment);
        _context.SaveChanges();
        return comment;
    }

    public List<Comment> GetByQuestionId(string questionId)
    {
        return _context.Comments
            .Where(c => c.QuestionId == questionId)
            .OrderBy(c => c.CreatedAt)
            .ToList();
    }

    public bool Delete(string id)
    {
        var comment = _context.Comments.Find(id);
        if (comment == null) return false;
        _context.Comments.Remove(comment);
        _context.SaveChanges();
        return true;
    }
}
